loki/transformations/remove_code.py

```diff
@@ -332,3 +337,68 @@ def visit_Import(self, o, **kwargs):
 
         rebuilt = tuple(self.visit(i, **kwargs) for i in o.children)
         return self._rebuild(o, rebuilt)
+
+
+def do_remove_unused_imports(routine):
```

Comment:
Is this not redundant to or at least overlapping in functionality with the `sanitise_imports` utility? I would prefer if we combined the two somehow.
https://github.com/ecmwf-ifs/loki/blob/83c9ebe002064e0544a4b3bf0261ca1bbe9323ee/loki/transformations/utilities.py#L281-L295